docs/User-Research/2021, Summer - Communicating Errors in TDP & Validations With Regional Staff.md

```diff
@@ -0,0 +1,370 @@
+
+- Giving grantee program directors a standardized way to submit a request for access for somebody on their team within TDP.
+- Requiring that non-director grantee staff have been authorized by their program director (via the above) to receive the access they're requesting.
+- Clearly displaying information about new users requesting access for the regional staff who will be approving them. (e.g email address, associated STT, associated/referring program director, etc...)
+
+---
+
+### We documented "error patterns"  that may be utilized throughout error guidance in TDP.
+
+**Naming Patterns**
+
+Plain Language Error Names
+
+- Affords glanceability (and potentially referenceability) to a user who might not be familiar with legacy error codes, or who might not have them committed to memory.
+
+Legacy Error Names
+
+- Largely for those either playing a role in designing their own grantee level systems or those already highly familiar with existing error codes. Draws a through line showing which legacy errors map to newer, plainer language TDP errors. Can be used alongside the newer error names.
+
+**Scope Patterns**
+
+Split Errors
+
+- There seems to be potential to make certain legacy TDRS errors more actionable by splitting them into several variations. One example being T1-515 (Update Indicator is not valid). In TDP we could provide several variations that better identify what's wrong with the update indicator and better contextualize what will happen to previously submitted data as a result of correcting it in different ways (e.g. D, U, N).
+
+**Guidance Patterns**
+
+"See More" Links
+
+- Provides additional guidance for errors that are much more involved to fix (e.g. "Other") or any other errors when lengthier guidance is warranted and useful. In the early versions of TDP this is likely less common, unless we prioritize the creation of new guidance content to supplement current error and transmission layout documents.
+
+Side-by-sides
+
+- Errors that compare data submitted against what was expected. Can potentially be blocked by data issues severe enough to block parsing as a whole. In cases where user input can't be predicted, the "Expected" side would be limited to an example or a breakdown of the relevant layout.
+
+Snippets
+
+- Overlaps with side by sides, snippets are simply excerpts or breakdowns of parts of the submitted file or parts of the expected layout. The distinction is simply that a snippet could be used with or without user input depending on the context (e.g. in guidance content that's accessible pre-submission or around submissions, not just specific errors during a specific submission)
+
+**Call to Action Patterns**
+
+Bypass
+
+- In-app ability to actively ignore and dismiss certain errors.
```

Comment:
is this about pre- or post-parsing errors? i dont think we want this capability for pre-parsing errors,. And yes, for post-parsing errors, OFA will determine what types of post-parsing errors can be "bypassed", as we have w/ TDRS.

Reply:
Yeah, this is for post parsing. A potential stopgap on the way to validation rule adjustments.